dist/znk.infra.js

```diff
@@ -379,6 +418,57 @@ angular.module('znk.infra.analytics').run(['$templateCache', function($templateC
 
 }]);
 
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.assignModule').service('UserAssignModuleService', [
+        'ZnkModuleService', 'ModuleResultsService', '$q',
+        function (ZnkModuleService, ModuleResultsService, $q) {
+            var userAssignModuleService = {};
+
+            userAssignModuleService.getUserAssignModules = function (userId) {
+                return ModuleResultsService.getUserModuleResultsGuids(userId).then(function (resultsGuids) {
+                    var moduleResults = {};
+                    var getProm = $q.when();
```

Comment:
its a trick to make sure $q.all will wait all the promises we create within angular.forEach